source/javascripts/controllers/_EnvVarsController.js.erb

```diff
@@ -186,6 +199,45 @@ angular.module("BitriseWorkflowEditor").controller("EnvVarsController", function
 		return errors.length > 0 ? stringService.errorMessageFromErrors(errors) : null;
 	};
 
+	viewModel.valueVisibilityGetterSetterForEnvVar = function(secret) {
+		return function(shouldShowValue) {
+			if (shouldShowValue !== undefined) {
+				if (shouldShowValue && !secret.customProperties.progress.isInProgress) {
+					showValueOfSecret(secret);
+				}
+
+				secret.shouldShowValue = shouldShowValue;
+			}
+
+			return secret.shouldShowValue;
+		};
+	};
+
+	function showValueOfSecret(secret) {
```

Comment:
Rename to `getSecretValue` - only nitpicking, since a few changes are already needed.